files/nrpe/check_horizon_ext

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
+#
+# Check if Horizon login works externally (outside of haproxy)
+#
+# Adapted from a script by https://ask.openstack.org/en/users/17403/amedeo-salvati/
+
+STATE_OK=0
+STATE_WARNING=1
+STATE_CRITICAL=2
+STATE_UNKNOWN=3
+
+# Defaults: Curl timeout plus warn/crit thresholds
+MAX_TIME=60
+WARN=30
+CRIT=45
+
+usage ()
+{
+  echo "Usage: $0 [OPTIONS]"
+  echo " -h               Get help"
+  echo " -U <username>    Username to use for Horizon login"
+  echo " -P <password>    Password to use for Horizon login"
+  echo " -R <URL>         Region URL (Keystone endpoint)"
+  echo " -H <host>        Base host part for curl call URLs"
+  echo " -I <project_id>  Project ID"
+  echo " -T <seconds>     Timeout"
+}
+
+cleanup ()
+{
+  rm -f $COOKIE_FILE
+  rm -f $CSV_FILE
+}
+
+if (($# == 0)); then
+  usage
+  exit 1
+fi
+
+while getopts 'hU:P:R:H:I:w:c:' OPTION
+do
+    case $OPTION in
+        h)
+            usage
+            exit ${STATE_UNKNOWN}
+            ;;
+        U)
+            export HZ_TEST_USERNAME=$OPTARG
+            ;;
+        P)
+            export HZ_TEST_PASSWORD=$OPTARG
+            ;;
+        R)
+            export HZ_TEST_REGION=$OPTARG
+            ;;
+        H)
+            export HZ_TEST_HOST=$OPTARG
+            ;;
+        I)
+            export HZ_TEST_PROJECT_ID=$OPTARG
+            ;;
+        T)
+            export MAX_TIME=$OPTARG
+            ;;
+        w)
+            export WARN=$OPTARG
+            ;;
+        c)
+            export CRIT=$OPTARG
+            ;;
+	*)
+            usage
+            exit ${STATE_UNKNOWN}
+            ;;
+    esac
+done
+shift "$((OPTIND-1))"
+
+CURL="/bin/curl"
+COOKIE_FILE=$(/usr/bin/mktemp)
+CSV_FILE=$(/usr/bin/mktemp)
```

Comment:
It would be cool, but totally not necessary to do `CSV_FILE=$(/usr/bin/mktemp --suffix -$(basename "$0"))`